app/src/org/commcare/android/resource/installers/FileSystemInstaller.java

```diff
@@ -213,6 +213,10 @@ public boolean upgrade(Resource r, AndroidCommCarePlatform platform) {
             String finalLocation = localDestination + "/" + filepart;
 
             if (!FileSystemUtils.moveFrom(localLocation, finalLocation, false)) {
+                Logger.log(LogTypes.TYPE_RESOURCES, "Failed to move resource " + r.getDescriptor() +
+                        " during upgrade from origin file path " + localLocation + " to  desination file path " +
+                        finalLocation + ". Origin file exists = " + new File(localLocation).exists()
+                        + " and Destination File exits = " + new File(finalLocation).exists());

```

Comment:
Separately trying to confirm a suspicion that this fails sometimes causing the update installation to get blocked.